### PR TITLE
optimized and cleaned up wire pods and cameras

### PR DESCRIPTION
**Pull request by Aws0mee**

### CODE REVIEW

**Comment by Divran** [comment]
You replaced tab indents with space indents, making it very difficult to see what actually changed, because the diffs are now entirely red or entirely green. Please undo that change, then I can look at this again.

**Comment by thegrb93** [comment]
Instead of `table.insert` and `table.RemoveByValue` you can just `pods[self] = true` and change your iterators to `for v in pairs(pods) do`

**Comment by thegrb93** [comment]
Also we loath huge whitespace changes mixed with subtle code changes.

**Comment by thegrb93** [comment]
> Apologies, I noticed some of the formatting was ugly in my opinion so I thought it wouldn't hurt to pretty print it.

It makes reviewing harder

File: lua/entities/gmod_wire_pod.lua [hunk 1]
```diff
@@ -1,649 +1,692 @@
...
-	self.Pod = nil
-	self:PlayerExited()
-	WireLib.SendMarks(self, {})
-	WireLib.TriggerOutput( self, "Entity", NULL )
-	self:ColorByLinkStatus(self.LINK_STATUS_UNLINKED)
-	return true
+    if IsValid(self.Pod) then
+        self.Pod:RemoveCallOnRemove("wire_pod_remove")
+    end
+	table.RemoveByValue(pods, self)
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_pod.lua:215]: ok if you remove this here, and then relink the ent, wouldn't that be a problem?.

Comment (thegrb93) [reply lua/entities/gmod_wire_pod.lua:215]: They're spaces instead of tabs. Needs to be reverted.

Comment (Divran) [reply lua/entities/gmod_wire_pod.lua:215]: also it's far more efficient to use a lookup table instead of removebyvalue

Comment (thegrb93) [reply lua/entities/gmod_wire_pod.lua:215]: Well, it's more efficient to use ipairs for the loops though.

Comment (Aws0mee) [reply lua/entities/gmod_wire_pod.lua]: > ok if you remove this here, and then relink the ent, wouldn't that be a problem?
> 
> The solution seems to be to move this to OnRemove.

I thought that function was only called by OnRemove, and I saw that all of the code was there so I was trying to be consistent. I moved that one function call to OnRemove instead like you suggested. :)

File: lua/entities/gmod_wire_pod.lua [hunk 2]
```diff
@@ -11,230 +11,231 @@ if CLIENT then
...
-	self:SetShowCursor( 0 )
+function ENT:Initialize()
+	self:PhysicsInit(SOLID_VPHYSICS)
+	self:SetMoveType(MOVETYPE_VPHYSICS)
+	self:SetSolid(SOLID_VPHYSICS)
+	self:SetUseType(SIMPLE_USE)
+
+	table.insert(pods, self)
+
+	local outputs = {"W", "A", "S", "D", "Mouse1", "Mouse2", "R", "Space", "Shift", "Zoom", "Alt", "TurnLeftKey (Not bound to a key by default. Bind a key to '+left' to use.\nOutside of a vehicle, makes the player's camera rotate left.)", "TurnRightKey (Not bound to a key by default. Bind a key to '+right' to use.\nOutside of a vehicle, makes the player's camera rotate right.)", "PrevWeapon (Usually bound to the mouse scroller, so will only be active for a single tick.)", "NextWeapon (Usually bound to the mouse scroller, so will only be active for a single tick.)", "Light", "X", "Y", "Z", "AimPos [VECTOR]", "Distance", "Bearing (If the 'Relative' input is non-zero, this will be relative to the vehicle.)", "Elevation (If the 'Relative' input is non-zero, this will be relative to the vehicle.)", "ThirdPerson", "Team", "Health", "Armor", "Active", "Entity [ENTITY]", "Driver [ENTITY]"}
```
Comment (thegrb93) [inline lua/entities/gmod_wire_pod.lua:132]: Not a fan of this change. It was better before

File: lua/entities/gmod_wire_rt_camera.lua [hunk 3]
```diff
@@ -4,201 +4,190 @@ ENT.Type = "anim"
...
+		local filteringFlag = 1 -- pointsample
+
+		if wire_rt_camera_filtering:GetInt() == 1 then
+			filteringFlag = 2 -- trilinear
+		elseif wire_rt_camera_filtering:GetInt() == 2 then
+			filteringFlag = 16 -- anisotropic
+		end
+
+		local isHDR = wire_rt_camera_hdr:GetInt() ~= 0
+		local rt = GetRenderTargetEx(CreateRTName(index), wire_rt_camera_resolution_w:GetInt(), wire_rt_camera_resolution_h:GetInt(), RT_SIZE_LITERAL, MATERIAL_RT_DEPTH_SEPARATE, filteringFlag + 256 + 32768, isHDR and CREATERENDERTARGETFLAGS_HDR or 0, isHDR and IMAGE_FORMAT_RGBA16161616 or IMAGE_FORMAT_RGB888)
```
Comment (thegrb93) [inline lua/entities/gmod_wire_rt_camera.lua:144]: This change is worse

File: lua/entities/gmod_wire_rt_camera.lua [hunk 4]
```diff
@@ -4,201 +4,190 @@ ENT.Type = "anim"
...
+			self:InitRTTexture()
+		else
+			ObservedCameras[self.ObservedCamerasIndex] = nil
+			self.ObservedCamerasIndex = nil
+			self.RenderTarget = nil
+		end
+	end
 
+	local function CreateRTName(index)
+		return "improvedrtcamera_rt_" .. tostring(index) .. "_" .. wire_rt_camera_filtering:GetString() .. "_" .. wire_rt_camera_resolution_h:GetString() .. "x" .. wire_rt_camera_resolution_w:GetString() .. (wire_rt_camera_hdr:GetInt() and "_hdr" or "_ldr")
```
Comment (thegrb93) [inline lua/entities/gmod_wire_rt_camera.lua:130]: This change is worse

File: lua/entities/gmod_wire_rt_camera.lua [hunk 5]
```diff
@@ -4,201 +4,190 @@ ENT.Type = "anim"
...
+	-- The callback will not be called clientside if the var is changed right after entity spawn.
+	if CLIENT and self:GetActive() then
+		self:ActiveChanged(nil, nil, true)
+	end
 end
 
-function ENT:Setup(default_fov) --(model, default_fov)
-    --self:SetModel(model or "models/maxofs2d/camera.mdl")
-    self:SetCamFOV(default_fov or 80)
+--(model, default_fov)
```
Comment (thegrb93) [inline lua/entities/gmod_wire_rt_camera.lua:31]: Delete comments

File: lua/entities/gmod_wire_pod.lua [hunk 6]
```diff
@@ -395,45 +411,63 @@ function ENT:TriggerInput( name, value )
...
-			pod:Fire("HandBrakeOff","1",0)
+			pod:Fire("TurnOn", "1", 0)
+			pod:Fire("HandBrakeOff", "1", 0)
 		end
 	elseif (name == "Damage Health") then
 		if not self:HasPly() or value <= 0 then return end
-		if (value > 100) then value = 100 end
-		self:GetPly():TakeDamage( value )
+
+		if (value > 100) then
```
Comment (thegrb93) [inline lua/entities/gmod_wire_pod.lua:425]: All these `if then` blocks look worse

File: lua/entities/gmod_wire_pod.lua [hunk 7]
```diff
@@ -11,230 +11,231 @@ if CLIENT then
...
-		if b then RestoreCursorPosition() else RememberCursorPosition() end
-		gui.EnableScreenClicker( b )
+		pnl:SetWorldClicker(b) -- this allows the cursor to move the player's eye
+
+		if b then
+			RestoreCursorPosition()
+		else
+			RememberCursorPosition()
+		end
+
```
Comment (thegrb93) [inline lua/entities/gmod_wire_pod.lua:101]: This change is worse

File: lua/entities/gmod_wire_pod.lua [hunk 8]
```diff
@@ -11,230 +11,231 @@ if CLIENT then
 	local HUDHidden = false
 	local savedHooks = nil
 	local toolgunHUDFunc = nil
-	local function blank() end
 
-	usermessage.Hook( "wire pod hud", function( um )
+	local function blank()
+	end
```
Comment (thegrb93) [inline lua/entities/gmod_wire_pod.lua:16]: This change is worse

**Comment by Divran** [comment]
@thegrb93 yep forgot to mention it but I've already talked to him and he mentioned he was going to create a new PR without any formatting changes. I also mentioned to him that many of the changes his editor made when he pressed the auto indent button are just straight up worse.

**Comment by TomyLobo** [comment]
I figured it was an auto formatter of some sort.
And a bad one at that.
It moved some comments from inside multi-line statements to after them, yikes.

**Comment by thegrb93** [comment]
Going to close since he's making a new pr

**Pull request closed without merging**